From e689673817bc9379e775ed0f670e133521dffe3e Mon Sep 17 00:00:00 2001
From: Christophe Roullier <christophe.roullier@st.com>
Date: Wed, 14 Jun 2023 13:02:03 +0200
Subject: [PATCH 528/529] net: stmmac: Disable mac_managed_pm phylink config

Due to regression in suspend/resume for stm32mp13 DK
(sometime no IP address after resume)
Put in false mac_managed_pm to indicate that MAC driver
do not manages PHY PM.

Fixes: f151c147b3af ("net: stmmac: Enable mac_managed_pm phylink config")

Signed-off-by: Christophe Roullier <christophe.roullier@st.com>
Change-Id: I700aac71f85335a722b5a6a77fdbc9359a8f31ff
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/312665
Reviewed-by: Alexandre TORGUE <alexandre.torgue@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Alexandre TORGUE <alexandre.torgue@foss.st.com>
---
 drivers/net/ethernet/stmicro/stmmac/stmmac_main.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/net/ethernet/stmicro/stmmac/stmmac_main.c
+++ b/drivers/net/ethernet/stmicro/stmmac/stmmac_main.c
@@ -1238,7 +1238,7 @@ static int stmmac_phy_setup(struct stmma
 	if (priv->plat->tx_queues_to_use > 1)
 		priv->phylink_config.mac_capabilities &=
 			~(MAC_10HD | MAC_100HD | MAC_1000HD);
-	priv->phylink_config.mac_managed_pm = true;
+	priv->phylink_config.mac_managed_pm = false;
 
 	phylink = phylink_create(&priv->phylink_config, fwnode,
 				 mode, &stmmac_phylink_mac_ops);
